Documentation/EmacsConfiguration.md

```diff
@@ -0,0 +1,61 @@
+# Emacs Project Configuration
+
+Emacs can be configured with `lsp-mode` and `clangd` to work well.
+
+### clangd
+
+The official clangd extension can be used for C++ comprehension. You
+can use the following `.clangd` file placed in the project root:
+
+```yaml
+CompileFlags:
+  CompilationDatabase: Build/i686
+  Add:
+    - "-D__serenity__"
+    - "-UNO_TLS"
+    - "-I/path/to/serenity/Toolchain/Local/i686/i686-pc-serenity/include/c++/11.2.0"
+    - "-I/path/to/serenity/Toolchain/Local/i686/i686-pc-serenity/include/c++/11.2.0/i686-pc-serenity"
+```
+
+You will obviously need to update `11.2.0` to whatever your version at
+the time is.
+
+Run cmake at least once for this to work. clangd has difficulty
+finding specific methods and types, especially with inheritance
+trees. Also, include errors are wrong in 90% of cases.
+
+### lsp-mode
+
+```lisp
+(use-package lsp-mode
+  :hook ((c++-mode) . lsp-deferred)
+  :commands lsp
+  :config
+  (setq lsp-clients-clangd-args '("-j=4" "-background-index" "--log=error" "--clang-tidy" "--enable-config"))
+  (setq lsp-clangd-binary-path "/usr/bin/clangd"))
+
+```
```

Comment:
This seems to set `lsp-clangd-binary-path` to the system one all the time. Is this intentional?

Reply:
I might have a misunderstanding but I believe there are 3 options

- System clangd (`/usr/bin/clangd`)
- Serenity once, if you build the clang toolchain. Someone showed me how recently, but since its not built by default and I'm fairly ignorant on the subject, I didn't want to recommend that.
- Auto-installed clangd from `lsp-mode`, (`~/.emacs.d/cache/lsp/clang/...`). I thought this would trip people up since its in a non standard location and if people ran `clangd` on the command line they would get a different install and possibly version than the one `lsp-mode` is running. 

I just picked the first one as it has the least amount of surprise. I'm totally open to suggestions for a better default, or to possibly even leave it out if you prefer.

Reply:
@sin-ack Any thoughts on which of these three you prefer? After we decide on this one I can also improve the other remaining issue you raised (explaining how to set a different binary path).

Reply:
I think it should read something like:

- By default, `lsp-mode` will use your system clangd. This is the easiest solution but your system clangd might be out of date
- Serenity can build you an LLVM toolchain with clangd (cc @ADKaster for details), which will provide you with better support. To do this, <insert instructions here> then use `.dir-locals.el` to set `lsp-clangd-binary-path` on `c++-mode`
- If you want lsp to manage your clangd instead, use `lsp-install-server`

So that we cover all choices.